pkg/providers/apisix/apisix_upstream.go

```diff
@@ -373,9 +383,19 @@ func (c *apisixUpstreamController) updateExternalNodes(ctx context.Context, au *
 	upsName := apisixv1.ComposeExternalUpstreamName(ns, name)
 	ups, err := c.APISIX.Cluster(clusterName).Upstream().Get(ctx, upsName)
 	if err != nil {
-		log.Errorf("failed to get upstream %s: %s", upsName, err)
-		c.RecordEvent(au, corev1.EventTypeWarning, utils.ResourceSyncAborted, err)
-		c.recordStatus(au, utils.ResourceSyncAborted, err, metav1.ConditionFalse, au.GetGeneration())
+		if err == apisix.ErrNotFound {
+			log.Debugw("upstream is not referenced",
+				zap.String("cluster", clusterName),
+				zap.String("upstream", upsName),
+			)
+			err = fmt.Errorf("%s", "upstream doesn't exist. It will be created after ApisixRoute is created referencing it.")
+			c.RecordEvent(au, corev1.EventTypeWarning, utils.ResourceSynced, err)
+			c.recordStatus(au, utils.ResourceSynced, err, metav1.ConditionFalse, au.GetGeneration())
+		} else {
+			c.RecordEvent(au, corev1.EventTypeWarning, utils.ResourceSyncAborted, err)
+			c.recordStatus(au, utils.ResourceSyncAborted, err, metav1.ConditionFalse, au.GetGeneration())
+			log.Errorf("failed to get upstream %s: %s", upsName, err)
+		}
```

Comment:
```suggestion
		if err == apisix.ErrNotFound {
			log.Debugw("upstream is not referenced",
				zap.String("cluster", clusterName),
				zap.String("upstream", upsName),
			)
			err = fmt.Errorf("%s", "upstream doesn't exist. It will be created after ApisixRoute is created referencing it.")
			c.RecordEvent(au, corev1.EventTypeWarning, utils.ResourceSynced, err)
			c.recordStatus(au, utils.ResourceSyncAborted, err, metav1.ConditionFalse, au.GetGeneration())
		} else {
			c.RecordEvent(au, corev1.EventTypeWarning, utils.ResourceSyncAborted, err)
			c.recordStatus(au, utils.ResourceSyncAborted, err, metav1.ConditionFalse, au.GetGeneration())
			log.Errorf("failed to get upstream %s: %s", upsName, err)
		}
```